File: OOPS/Python/abstraction.py
from abc import ABC
class Shape(ABC):
    def area(self):
        pass
    def perimeter(self):
        pass
class Square(Shape):
    def __init__(self, side):
        self.side = side
    def area(self):
        return self.side * self.side
    def perimeter(self):
        return 4 * self.side
if __name__=="__init__":
    square = Square(10)
    print(square.area())
    print(square.perimeter())

File: OOPS/Abstraction.java
interface Shape{
    abstract double area();
    abstract double perimeter();
}
class Square implements Shape{
    private double side;
    Square(double side){
        this.side = side;
    }
    @Override
    public double area(){
        return side*side;
    }
    @Override
    public double perimeter(){
        return 4*side;
    }
}
class Abstraction {
    public static void main(String[] args) {
        Square s = new Square(5);
        System.out.println("Area of square is: "+s.area());
        System.out.println("Perimeter of square is: "+s.perimeter());
    }
}
